bundle/bundle-writer.go

```diff
@@ -0,0 +1,352 @@
+package bundle
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cloud66-oss/starter/common"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type ManifestBundle struct {
+    Version        string                   `json:"version"`
+    Metadata       *Metadata                `json:"metadata"`
+    Uid            string                   `json:"uid"`
+    Name           string                   `json:"name"`
+    StencilGroups  []*BundleStencilGroup    `json:"stencil_groups"`
+    BaseTemplates  []*BundleBaseTemplates   `json:"base_template"`
+    Policies       []*BundlePolicy          `json:"policies"`
+    Tags           []string                 `json:"tags"`
+    HelmReleases   []*BundleHelmRelease     `json:"helm_releases"`
+    Configurations []string                 `json:"configuration"`
+}
+
+
+type BundleHelmRelease struct {
+    Name             string `json:"repo"`
+    Version          string `json:"version"`
+    RepositoryURL    string `json:"repository_url"`
+	ValuesFile       string `json:"values_file"`
+}
+
+type BundleConfiguration struct {
+    Repo   string `json:"repo"`
+    Branch string `json:"branch"`
+}
+
+type BundleBaseTemplates struct {
+    Repo     string `json:"repo"`
+    Branch   string `json:"branch"`
+    Stencils []*BundleStencil `json:"stencils"`
+}
+
+type Metadata struct {
+    App         string     `json:"app"`
+    Timestamp   time.Time  `json:"timestamp"`
+    Annotations []string   `json:"annotations"`
+}
+
+type BundleStencil struct {
+    Uid              string   `json:"uid"`
+    Filename         string   `json:"filename"`
+    TemplateFilename string   `json:"template_filename"`
+    ContextID        string   `json:"context_id"`
+    Status           int      `json:"status"`
+    Tags             []string `json:"tags"`
+    Sequence         int      `json:"sequence"`
+}
+
+type BundleStencilGroup struct {
+    Uid  string   `json:"uid"`
+    Name string   `json:"name"`
+    Tags []string `json:"tags"`
+}
+
+type BundlePolicy struct {
+    Uid      string   `json:"uid"`
+    Name     string   `json:"name"`
+    Selector string   `json:"selector"`
+    Tags     []string `json:"tags"`
+}
+
+
+func CreateSkycapFiles(outputDir string,
+						templateRepository string,
+						branch string,
+						pack_name string,
+						githubURL string,
+						services []*common.Service,
+						databases []common.Database) error {
+
+	if templateRepository == "" {
+		//no stencil template defined for this pack, print an error and do nothing
+		fmt.Printf("Sorry but there is no stencil template for this language/framework yet\n")
+		return nil
+	}
+	//Create .bundle directory structure if it doesn't exist
+	skycapFolder := filepath.Join(outputDir, "skycap")
```

Comment:
this should be done using OS temp directory / temp file APIs: https://yourbasic.org/golang/temporary-file-directory/

Reply:
I should create only the bundle base folder with the temp (this way will be unique) or all of them?

Reply:
If you need to have a temporary folder anywhere in the system which will be removed after you're done, it should be in the Temp folder, using the golang OS temp folder functions.